remote_config/src/config_update_listener_registration.h

```diff
@@ -0,0 +1,48 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FIREBASE_REMOTE_CONFIG_SRC_CONFIG_UPDATE_LISTENER_REGISTRATION_H_
+#define FIREBASE_REMOTE_CONFIG_SRC_CONFIG_UPDATE_LISTENER_REGISTRATION_H_
+
+#include "remote_config/src/common.h"
+
+namespace firebase {
+namespace remote_config {
+
+//  Calling Remove stops the listener from receiving
+//  config updates and unregisters itself. If remove is called and no other
+//  listener registrations remain, the connection to the Remote Config backend
+//  is closed. Subsequently calling AddOnConfigUpdate will re-open the
+//  connection.
+class ConfigUpdateListenerRegistration {
```

Comment:
Move this file to remote_config/src/include/firebase/remote_config/
Since this is a new public header, also add it to remote_config_HDRS in /app/CMakeLists.txt 
(I'm not sure exactly how it gets used, but we enumerate all public headers there)

This class is part of the public RemoteConfig API, so this header file should be under src/include/firebase. The pattern that we follow is that there is one main public header and then a subfolder that contains all other public headers and the main header includes the other public headers.

The corresponding source file can stay where you have it.

Reply:
Updated CMake file and moved to subfolder in include/firebase